index.js

```diff
@@ -37,31 +57,53 @@ function validate(word) {
 }
 
 function stonksify(word) {
-  //If a word starts with a c replace it with a k
-  //Example: cat => kat
+  let modifiedCharacters = Array.apply(null, Array(word.length)).map(
+    function() {
+      return 0;
+    }
+  );
+
   word = word.toLowerCase();
-  if (word.indexOf("c") === 0) {
-    word = word.replace(/^c/, "k");
-    return word;
-  }
+  for (let i = 0; i < stonksifyRules.length; i++) {
+    let toReplace = new RegExp(stonksifyRules[i][0]);
+    let replaceWith = stonksifyRules[i][1];
+    let newWord = word.replace(toReplace, replaceWith);
+    console.log(
+      "Stonksify rule " +
+        i +
+        ": replace regexp '" +
+        toReplace +
+        "' with '" +
+        replaceWith +
+        "'."
+    );
 
-  //If a word has one c replace it with an n
-  //Example: stocks => stonks
-  if (word.split("c").length - 1 === 1) {
-    word = word.replace(/c/g, "n");
-    return word;
-  }
+    let changedCharactersModified = false;
+    let changedCharactersIndexesChanged = [];
+    for (let ii = 0; ii < modifiedCharacters.length; ii++) {
```

Comment:
Why not just i?

Reply:
used to be inside another loop before i took out the double rule application check